src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
I don't think premature optimization applies on the critical path.

As for having to reimplement `Sugar.Names`, you're basically just mapping over the declaration refs part of the modules, can't you compose that with the existing desugarings or if that's impossible make it configurable? I don't think we should be doing more work in the compile process because it's more convenient for documentation generation.